idaes_examples/notebooks/docs/scaling/scaler_workshop.ipynb

```diff
@@ -0,0 +1,1944 @@
+    "\n",
+    "For the equilibrium reactor, we have the following variables we need to scale:\n",
+    "\n",
+    "1. Rate-based reaction extent and generation terms\n",
+    "2. Equilibrium-based reaction extent and generation terms\n",
+    "3. Inherent reaction extent and generation terms\n",
+    "4. Phase equilibrium generation terms\n",
+    "5. Energy balance heat term\n",
+    "6. Energy balance heats of reaction\n",
+    "7. Pressure drop\n",
+    "\n",
+    "Many of these are hard to know a priori - anything related to a reaction is very hard to know without knowing the reaction behavior. Considering that the equilibrium reactor is modular, we have little to no way of knowing these in the general case (and even in the specific test case it is hard enough). We can assume that the reaction package will scale all of its variables (i.e., rate and equilibrium constants, and reaction rates), however it is hard to project these to unit model scaling.\n",
+    "\n",
+    "For a CSTR we can say that ``extent = volume*rate`` and thus estimate scaling, but this does not work for equilibrium systems where 1) volume is undefined, 2) reaction rate at the outlet state is being driven to zero to satisfy equilibrium, and 3) extent is solved implicitly to satisfy the need for reaction rate to equal zero.\n",
+    "\n",
+    "Considering that a bad guess is often worse than no guess, we will not scale these right now - it is important to remember that our goal is to improve the overall scaling so if we do not know how to scale something it is generally best to leave it unscaled. We might come back to these later if necessary, but for now we will leave these either for the user to provide based on knowledge of their system, or for automated fill-in using some autoscaler.\n",
+    "\n",
+    "For the heat and deltaP terms, these are dependent on extensive variables in each case study and we have no way of knowing their exact values. However, we can probably take a good guess at order-of-magnitude using engineering knowledge; heat duties are generally approximately one order of magnitude smaller than the enthalpy flows,\n",
+    "and pressure drops are generally on the order of 0.1 bar.\n",
+    "\n",
+    "To apply scaling for the pressure drop term, we can make use of the ``scale_variable_by_units`` method in ``CustomscalerBase``. This method looks up the units of measurement for the variable, and then loops in the class attribute ``UNIT_SCALING_FACTORS`` dictionary to find a equivalent unit for the quantity of interest and an associated scaling factor. If a scaling factor is found, it is converted as necessary; e.g., in this case pressure is defined in ``Pa`` but we can set the default scaling factor in ``bar`` and it will be converted as appropriate. The code required to do this is below.\n",
```

Comment:
```suggestion
    "To apply scaling for the pressure drop term, we can make use of the ``scale_variable_by_units`` method in ``CustomscalerBase``. This method looks up the units of measurement for the variable, and then loops in the class attribute ``UNIT_SCALING_FACTORS`` dictionary to find an equivalent unit for the quantity of interest and an associated scaling factor. If a scaling factor is found, it is converted as necessary; e.g., in this case pressure is defined in ``Pa`` but we can set the default scaling factor in ``bar`` and it will be converted as appropriate. The code required to do this is below.\n",
```